pkg/kaniko/builder.go

```diff
@@ -118,12 +117,11 @@ func CreateBuilder(ctx context.Context, cfg Config, shell executor.ShellExecutor
 			logger.Fatalf("cannot create kaniko kubernetes executor: %v", err)
 		}
 
-		s3Cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(cfg.Context.S3.Region))
+		s3, err := buildcontext.NewS3Uploader(ctx, cfg.Context.S3.Region, cfg.Context.S3.Bucket)
```

Comment:
fail fast en cas de mauvaise configuration. Je te propose d'ajouter ces verifications:


```suggestion
		s3, err := buildcontext.NewS3Uploader(ctx, cfg.Context.S3.Region, cfg.Context.S3.Bucket)
				var uploader buildcontext.FileUploader

		switch {
		case cfg.Context.Azure.AccountName != "" && cfg.Context.S3.Bucket != "":
			logger.Fatalf("only one of Azure or S3 can be configured for build context upload")
		case cfg.Context.Azure.AccountName != "":
			uploader, err = buildcontext.NewAzureUploader(cfg.Context.Azure.AccountName, cfg.Context.Azure.Container)
		case cfg.Context.S3.Bucket != "":
			uploader, err = buildcontext.NewS3Uploader(ctx, cfg.Context.S3.Region, cfg.Context.S3.Bucket)
		default:
			logger.Fatalf("either Azure or S3 must be configured for build context upload")
		}
```

Reply:
Le builder kaniko étant déprécié et bientôt supprimé, je ne pense pas utile de modifier le code pour le support d'Azure. Qu'en penses-tu ?